File: java/serpro/src/test/java/com/trainning/serpro/StudentRepositoryTest.java
package com.trainning.serpro;

import java.sql.SQLException;
import java.util.ArrayList;
import org.junit.jupiter.api.AfterAll;
import static org.junit.jupiter.api.Assertions.assertEquals;
import org.junit.jupiter.api.BeforeEach;
import org.junit.jupiter.api.BeforeAll;
import org.junit.jupiter.api.Test;

public class StudentRepositoryTest {

    static StudentRepository studentRepository;
    static String studentName;
    static String studentNameModified;

    @BeforeAll
    public static void setUpClass() throws SQLException {
        studentRepository = new StudentRepository();
        studentName = "Joãozinho";
        studentNameModified = "Joãozinho da Silva";
    }

    @AfterAll
    public static void tearDownClass() {
        studentRepository = null;
        System.gc();
    }

    @BeforeEach
    public void setUp() throws SQLException {
        studentRepository.clear();
    }

    @Test
    public void testCreate() throws SQLException {
        Student student = studentRepository.create(studentName);
        assertEquals(1, student.getId(), "O Id do estudante deve ser 1");
        assertEquals(studentName, student.getNome(), "O Nome do estudante deve ser " + studentName);
    }

    @Test
    public void testUpdate() throws SQLException {
        this.testCreate();
        Student student = studentRepository.update(1, studentNameModified);
        assertEquals(1, student.getId(), "O Id do estudante deve ser 1");
        assertEquals(studentNameModified, student.getNome(), "O Nome do estudante deve ser " + studentNameModified);
    }

    @Test
    public void testList() throws SQLException {
        this.testCreate();
        ArrayList<Student> students = studentRepository.list();
        assertEquals(1, students.get(0).getId(), "O Id do estudante na posição 0 deve ser 1");
        assertEquals(studentName, students.get(0).getNome(), "O Nome do estudante na posição 0 deve ser " + studentName);
    }

    @Test
    public void testDelete() throws SQLException {
        this.testList();
        studentRepository.delete(1);
        ArrayList<Student> students = studentRepository.list();
        assertEquals(0, students.size(), "O tamanho da lista deve ser 0");
    }
}

File: python/src/StudentRepositoryTest.py
import unittest
from StudentRepository import StudentRepository

studentName = "Joãozinho"
studentNameModified = "Joãozinho da Silva"
studentRepository = StudentRepository()

class StudentRepositoryTest(unittest.TestCase):
    def testClear(self):
        studentRepository.clear()

    def testCreate(self):
        student = studentRepository.create(studentName)
        self.assertEqual(studentName, student.getNome(), f"O nome do estudante deveria ser: {studentName}")

    def testUpdate(self):
        studentRepository.clear()
        self.testCreate()
        student = studentRepository.update(1,studentNameModified)
        self.assertEqual(studentNameModified, student.getNome(), f"O nome do estudante deveria ser: {studentNameModified}")

    def testList(self):
        studentRepository.clear()
        self.testCreate()
        students = studentRepository.list()
        self.assertEqual(studentName, students[0].getNome(), f"O nome do estudante deveria ser: {studentName}")

    def testUpdate(self):
        self.testList()
        studentRepository.delete(1)
        students = studentRepository.list()
        self.assertEqual(0, len(students), f"A lista de estudantes deveria estar vazia")

if __name__ == "__main__":
    unittest.main()